File: server/server-rest/src/main/scala/uk/co/stevewellington/shoppinglist/JettyLauncher.java
package uk.co.stevewellington.shoppinglist;

import com.google.inject.servlet.GuiceFilter;
import org.eclipse.jetty.server.Server;
import org.eclipse.jetty.servlet.ServletContextHandler;

import javax.servlet.DispatcherType;

import static java.util.EnumSet.allOf;

class JettyLauncher {

    static void main(String[] args) throws Exception {
        Server server = new Server(8080);
        ServletContextHandler handler = new ServletContextHandler(server, "/", ServletContextHandler.SESSIONS);
        handler.addFilter(GuiceFilter.class, "/*", allOf(DispatcherType.class));
        handler.addEventListener(new MyGuiceServletConfig());
        server.setHandler(handler);
        server.start();
        server.join();
    }
    
}

File: server/server-rest/src/main/scala/uk/co/stevewellington/shoppinglist/JettyLauncher.scala
package uk.co.stevewellington.shoppinglist

import java.util.EnumSet._
import javax.servlet.DispatcherType

import com.google.inject.servlet.GuiceFilter
import org.eclipse.jetty.server.Server
import org.eclipse.jetty.servlet.ServletContextHandler

object JettyLauncher {

  val portNumber = 8080
  val server = new Server(portNumber)
  val handler = new ServletContextHandler(server, "/", ServletContextHandler.SESSIONS)

  handler.addFilter(classOf[GuiceFilter], "/*", allOf(classOf[DispatcherType]))
  handler.addEventListener(new MyGuiceServletConfig)
  server.setHandler(handler)

  def main(args: Array[String]) {
    server.start()
    server.join()
  }

}
